flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java

```diff
@@ -188,6 +207,21 @@ CompletableFuture<String> triggerSavepoint(
      */
     CompletableFuture<Long> triggerCheckpoint(JobID jobId, CheckpointType checkpointType);
 
+    /**
+     * Triggers a detached savepoint for the job identified by the job id. The savepoint will be
+     * written to the given savepoint directory, or {@link
+     * org.apache.flink.configuration.CheckpointingOptions#SAVEPOINT_DIRECTORY} if it is null.
+     * Notice that: detach savepoint will return with a savepoint trigger id instead of the path
+     * future, that means the client will return very quickly.
+     *
+     * @param jobId job id
+     * @param savepointDirectory directory the savepoint should be written to
+     * @param formatType a binary format of the savepoint
+     * @return The savepoint trigger id
+     */
+    CompletableFuture<String> triggerDetachSavepoint(
```

Comment:
Let's use 'triggerDetachedSavepoint' aligned with 'stopWithDetachedSavepoint'.
You could check other places about 'detach savepoint'.